wolfcrypt/src/asn.c

```diff
@@ -8099,16 +8099,22 @@ int ParseCertRelative(DecodedCert* cert, int type, int verify, void* cm)
             WOLFSSL_MSG("About to verify certificate signature");
 
             if (cert->ca) {
-                /* Check if cert is CA type and has path length set */
+                /* Check if cert is CA type and signer has path length set */
                 if (cert->isCA && cert->ca->pathLengthSet) {
-                    /* Check root CA (self-signed) has path length > 0 */
-                    if (cert->selfSigned) {
-                        if (cert->ca->pathLength != 0) {
-                           WOLFSSL_MSG("Root CA with path length > 0");
+                    /* Check if signer is root CA (self-signed) */
+                    if (cert->ca->selfSigned) {
+                        /* Root CA as signer:
+                         * Must have path length > 0 to sign another CA
+                         * If path length == 0 can only sign an end entity
+                         *   certificate, not intermediate CA
+                         */
+                        if (cert->ca->pathLength == 0) {
```

Comment:
The previous check of patchLength != 0 seemed to resolve a wpa_supplicant issue.  If this fixes some other issue how do we retain compatibility with original fix?

Reply:
The wpa_supplicant change was added in PR #1445 (3/21/18). I refactored the code in PR #1934. I talked with Sean and the changes in 1445 were made for a test case where a root CA certificate has pathLen == 0 and the test case requires this to fail.

My interpretation is a self signed CA cert can have a pathLen of 0, but in that case can only sign a cert, not an intermediate CA. Sean reviewed and said `Looking at the RFC I can't see a reason why the root has to be path length 0 but the wpa_supplicant tests require it to be so.`

So my resolution is to add back this special case for the supplicant only.